packages/test-utils/src/reporter.ts

```diff
@@ -41,14 +41,34 @@ export interface TestIdContext {
   attempt: number;
 }
 
+export type UploadStatusThreshold = "all" | "failed-and-flaky" | "failed";
+
+type UploadStatusThresholdInternal = UploadStatusThreshold | "none";
+
+export type UploadOption =
+  | boolean
+  | {
+      minimizeUploads?: boolean;
+      statusThreshold?: UploadStatusThreshold;
+    };
```

Comment:
I just realized that Playwright has somewhat similar options for videos/traces/screenshots:
https://github.com/microsoft/playwright/blob/664c4dd44297fc8147d10734b617801a6c464609/packages/playwright/types/test.d.ts#L5130-L5132

Maybe there would be value in aligning our functionality with theirs here

Reply:
Interesting...so like `uploadMode` rather than `statusThreshold`?

I don't know if this feels like it should be a goal necessarily, because it might end up being difficult to map shared concepts like this (in `test-utils`) to Playwright and Cypress _both_ in a way that lines up with their own naming conventions, and changing it to only match one seems odd (or at least, there's no clear "winner" to me in between the two).

I think I'm in favor of sticking with the names we settled on, but I'm totally open for discussing it and reconsidering.